Spigot-API-Patches/0238-Add-damaged-slot-to-PlayerItemDamageEvent.patch

```diff
@@ -0,0 +1,54 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: MTM123 <matiss234@gmail.com>
+Date: Sun, 9 Jun 2019 12:45:49 +0300
+Subject: [PATCH] Add damaged slot to PlayerItemDamageEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/player/PlayerItemDamageEvent.java b/src/main/java/org/bukkit/event/player/PlayerItemDamageEvent.java
+index 2d049633998b3965ce0f4a6a4cea3169da0d5658..41e4a75ab346ee442a5d72d12ef3394b797dcf50 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerItemDamageEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerItemDamageEvent.java
+@@ -15,14 +15,25 @@ public class PlayerItemDamageEvent extends PlayerEvent implements Cancellable {
+     private static final HandlerList handlers = new HandlerList();
+     private final ItemStack item;
+     private int damage;
++    private final int slot; //Paper - add damaged slot
+     private boolean cancelled = false;
+ 
+     public PlayerItemDamageEvent(@NotNull Player player, @NotNull ItemStack what, int damage) {
+         super(player);
+         this.item = what;
+         this.damage = damage;
++        this.slot = -1;
+     }
+ 
++    //Paper start
++    public PlayerItemDamageEvent(@NotNull Player player, @NotNull ItemStack what, int damage, int slot) {
++        super(player);
++        this.item = what;
++        this.damage = damage;
++        this.slot = slot;
++    }
++    //Paper end
++
+     /**
+      * Gets the item being damaged.
+      *
+@@ -46,6 +57,17 @@ public class PlayerItemDamageEvent extends PlayerEvent implements Cancellable {
+         this.damage = damage;
+     }
+ 
++    //Paper start
++    /**
++     * Gets the slot where the item took durability damage
++     *
```

Comment:
Maybe add an example or how this slot can be used: how to get the ItemStack at this slot from a Player instance. (But that's a stupid use case since the ItemStack is directly exposed in this method.)